docs/releases.md

```diff
@@ -0,0 +1,25 @@
+# Tobira versions
+
+(For context, please see our [versioning policy](./versioning-policy.md)!)
+
+This document lists all Tobira releases with the kinds of breaking changes done in each release.
```

Comment:
Can we transfer this information to the releases itself?

- This document seems very hard to automate
- It's hard to release with the release since the releases are linked, but you have to prepare this before the release.
- I can't just go to the release and find all necessary information there. I have to check another place. Something I surely forget as soon as I close this page ;-P

You could reproduce the same information by just providing the last breaking change at a release. For example, the `2.1-bear` release notes could have a line like this:

- Last breaking configuration change: `2.0-bear`

That way, you know you need to do something if you upgrade from a < 2 version.

You also could provide this as a table:

List of breaking changes. If you upgrade from an earlier version than listed here, make sure to check migration path documentation.

| Component | Last Break |
| -------- | -------------- |
| Configuration | `2.0-bear` |
| CLI | `2.0-bear` |
| MeliSearch | `1.2-antilope` |

Reply:
Just to be clear: the release notes would absolutely list all breaking changes (with additional information about them). 

I don't really mind your suggestion, but I still prefer the overview table I proposed as I think it gives a more useful overview and has some features that your suggestion does not have. For example, it's very easy and fast to see how many breaking changes happened in between two given versions. 

And I don't really get your arguments against the table.

> This document seems very hard to automate

I don't really see why this would be particularly harder to automate than other stuff. 

> It's hard to release with the release since the releases are linked, but you have to prepare this before the release.

That's not really a problem I think. A dead link for a few minutes is not a problem I'd worry about :shrug: 

> I can't just go to the release and find all necessary information there. I have to check another place.

All information is in the release notes. You can still just go through all release notes from your current version to the newest one and you would know about all breaking changes. The table I proposed is just an additional overview. And we can of course easily link to it from every release.

---

So ye, I don't really see a reason to change this.

Reply:
I don't really have a hard opinion. I thought the table was neat. And yes, we should definitely have more detailed info about breaking changes per release, independently of whether or not we implement this table. I also don't see a big problem with it, though. Most of the work necessary for automating it will be done anyway, no matter the format of the result of the automation, I think. :thinking:

Reply:
Tangentially this makes me wonder, though: There is no Changelog file in this PR. Do we not want that? That is, do we want the Changelog to just live on the GitHub releases page? :thinking:

Reply:
@JulianKniephoff I would say that we only have the changelog at the releases. I don't really wanna duplicate it into a `CHANGELOG.md` or something like that.

Reply:
From an automation point of view, it's much easier to generate a new document (release) than to modify an old one, which may even have been modified manually. It's true that, with the table, can can quickly see how many versions had breaking changes between two versions, but the important information is really that there is a breaking change, I think.

Am I dead set against this table? No. I just don't think it's very useful and a hassle to maintain for years to come.

One thing I really dislike though is your new idea of listing the same information at several places. If we maintain this table, I wouldn't put the same information on the releases. Link this table. The same goes for a separate changelog. I don't think we want to maintain that, given that we already have this information somewhere else.

Really, I'm just thinking that if, e.g. next year, I would take over the project – or even just a few releases – how can I make my experience as painless as possible. Adding steps, putting additional information all over the place, is likely something I would find annoying.

So, please, let's keep these to a minimum, but keep your table if you want :D

Reply:
In order to no longer block on this and just get this over with, I removed the table for now. As long as we don't wait for too long, we can still add it later. See the commit removing the table for more information.